src/main/java/org/pac4j/springframework/security/authentication/ClientAuthenticationToken.java

```diff
@@ -97,10 +97,10 @@ public Object getPrincipal() {
 
     @Override
     public void eraseCredentials() {
-        credentials.clear();
-        if (userProfile != null) {
-            userProfile.clear();
-        }
+        //credentials.clear();
```

Comment:
`credentials = null;` and then, if user profile is not, call the `clearSensitiveData` method.